src/main/java/net/coderbot/iris/gl/IrisRenderSystem.java

```diff
@@ -130,20 +134,40 @@ public static void detachShader(int program, int shader) {
 		GL30C.glDetachShader(program, shader);
 	}
 
-	public static void bindImageTexture(boolean useExt, int unit, int texture, int level, boolean layered, int layer, int access, int format) {
+	public static void bindImageTexture(int unit, int texture, int level, boolean layered, int layer, int access, int format) {
 		RenderSystem.assertThread(RenderSystem::isOnRenderThreadOrInit);
-		if (useExt) {
-			EXTShaderImageLoadStore.glBindImageTextureEXT(unit, texture, level, layered, layer, access, format);
+			if (GL.getCapabilities().OpenGL42) {
+				GL42C.glBindImageTexture(unit, texture, level, layered, layer, access, format);
+			} else {
+				EXTShaderImageLoadStore.glBindImageTextureEXT(unit, texture, level, layered, layer, access, format);
+			}
+	}
+	
+	public static int getMaxImageUnits() {
+		if (GL.getCapabilities().OpenGL42) {
+			return GlStateManager._getInteger(GL42C.GL_MAX_IMAGE_UNITS);
+		} else if (GL.getCapabilities().GL_EXT_shader_image_load_store) {
+			return EXTShaderImageLoadStore.GL_MAX_IMAGE_UNITS_EXT;
```

Comment:
This should be a _getInteger call as well. It was a bug from the image load store PR.

Reply:
:+1:

Reply:
fixed